contracts/examples/guards/DelegateCallTransactionGuard.sol

```diff
@@ -2,13 +2,15 @@
 pragma solidity >=0.7.0 <0.9.0;
 
 import {Enum} from "../../libraries/Enum.sol";
-import {BaseGuard} from "../../base/GuardManager.sol";
+import {BaseGuard, Guard} from "../../base/GuardManager.sol";
+import {BaseModuleGuard, IModuleGuard} from "../../base/ModuleManager.sol";
+import {IERC165} from "../../interfaces/IERC165.sol";
 
 /**
  * @title DelegateCallTransactionGuard - Limits delegate calls to a specific target.
  * @author Richard Meissner - @rmeissner
  */
-contract DelegateCallTransactionGuard is BaseGuard {
+contract DelegateCallTransactionGuard is BaseGuard, BaseModuleGuard {
```

Comment:
If we rename `BaseGuard` to `BaseTransactionGuard`, then we can add a:

```solidity
abstract contract BaseGuard is ITransactionGuard, IModuleGuard {

}
```

for guards that are both.

Reply:
https://github.com/safe-global/safe-smart-account/pull/758/commits/3e4f790849556a5bcc815beb3936af16a3e45ce2